addons/controllers/packageinstallstatus_controller_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kappctrlv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	kapppkgiv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+	kapppkgv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	"github.com/vmware-tanzu/carvel-secretgen-controller/pkg/apis/secretgen/v1alpha1"
+	versions "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/versions/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/types"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+	"github.com/vmware-tanzu/tanzu-framework/addons/testutil"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+)
+
+var _ = Describe("PackageInstallStatus Reconciler", func() {
+	var (
+		antreaPkg               = "antrea.tanzu.vmware.com.1.2.3--vmware.1-tkg.2"
+		antreaPkgRefName        = "antrea.tanzu.vmware.com"
+		antreaPkgShortName      = "antrea"
+		clusterResourceFilePath string
+		clusterNameMng          = "test-mng"
+		clusterNamespaceMng     = "tkg-system"
+		clusterNameWlc          = "test-wlc"
+		clusterNamespaceWlc     = "test-ns-wlc"
+		kappPkgRefName          = "kapp-controller.tanzu.vmware.com"
+		mngAntreaObjKey         = client.ObjectKey{Namespace: constants.TKGSystemNS, Name: antreaPkgShortName}
+		wlcAntreaObjKey         = client.ObjectKey{Namespace: constants.TKGSystemNS, Name: util.GeneratePackageInstallName(clusterNameWlc, antreaPkgRefName)}
+		wlcKappObjKey           = client.ObjectKey{Namespace: clusterNamespaceWlc, Name: util.GeneratePackageInstallName(clusterNameWlc, kappPkgRefName)}
+	)
+
+	JustBeforeEach(func() {
+		By("Creating resources")
+		f, err := os.Open(clusterResourceFilePath)
+		Expect(err).ShouldNot(HaveOccurred())
+		defer f.Close()
+		Expect(testutil.CreateResources(f, cfg, dynamicClient)).Should(Succeed())
+
+		By("Installing antrea into management cluster")
+		testInstallPackage(clusterNameMng, antreaPkg, antreaPkgShortName, constants.TKGSystemNS)
+		testUpdatePkgInstallStatus(mngAntreaObjKey, kappctrlv1alpha1.ReconcileSucceeded)
```

Comment:
question: `testUpdatePkgInstallStatus` confuses me a little bit until I dig into the function. Was thinking why we `test` something here in `JustBeforeEach` block.

Reply:
And the same feeling about `testInstallPackage` as well. I originally thought "is it a test assertion" ?

Reply:
hmm, I used ```test``` prefix for all the utility test functions so that the function names can not collide in the future with production utility functions. I can drop the prefix though if it sounds confusing.